app/main.py

```diff
@@ -1,2 +1,163 @@
+        index = self._calculate_index(key)
+
+        if not self._hash_table[index]:
+            if self._size + 1 >= self._threshold:
+                self._resize_hash_table()
+                index = self._calculate_index(key)
+            self._hash_table[index] = Dictionary.Node(key, hash(key), value)
+            self._size += 1
+        else:
+            self._hash_table[index].value = value
+
+    def __delitem__(self, key: Hashable) -> None:
+        index = self._calculate_index(key)
+
+        if not self._hash_table[index]:
+            raise KeyError(f"'{key}'")
+
+        self._hash_table[index] = None
+        self._size -= 1
+
+    def __iter__(self) -> Dictionary:
+        def generator() -> Hashable:
+            for item in self._hash_table:
+                if item:
+                    yield item.key
+
+        self._generator = generator()
+        return self
+
+    def __next__(self) -> Hashable:
+        return next(self._generator)
+
+    def __len__(self) -> int:
+        return self._size
+
+    def __str__(self) -> str:
+        return str(self._hash_table)
+
+    def _resize_hash_table(self) -> None:
+        previous_hash_table = self._hash_table
+
+        self.__init__(
+            self._capacity * self._capacity_multiplier,
+            self._load_factor,
+            self._capacity_multiplier
+        )
+
+        for node in previous_hash_table:
+            if node:
+                self[node.key] = node.value
+
+    def clear(self) -> None:
+        self._size = 0
+        self._hash_table = [None] * self._capacity
+
+    def get(self, key: Hashable, default_value: Optional[Any] = None) -> Any:
+        try:
+            return self[key]
+        except KeyError:
+            if not default_value:
+                raise
+            return default_value
+
+    def pop(self, key: Hashable, default_value: Optional[Any] = None) -> Any:
+        try:
+            item = self[key]
+            del self[key]
+            return item.value
+        except KeyError:
+            if not default_value:
+                raise
+            return default_value
+
+    def keys(self) -> list[Hashable]:
+        return [
+            item.key
+            for item in self._hash_table
+            if isinstance(item, Dictionary.Node)
+        ]
+
+    def values(self) -> list[Any]:
+        return [
+            item.value
+            for item in self._hash_table
+            if isinstance(item, Dictionary.Node)
+        ]
+
+    def items(self) -> list[tuple[Hashable, Any]]:
+        return [
+            (item.key, item.value)
+            for item in self._hash_table
+            if isinstance(item, Dictionary.Node)
+        ]
+
+    def update(
+            self,
+            other: Dictionary | Iterable[Hashable, Any] = None
+    ) -> None:
+        if isinstance(other, Dictionary):
+            for key, value in other.items():
+                self[key] = value
+        elif isinstance(other, Iterable):
```

Comment:
Here you can omit this check. Try to unpack values like you did below and python will throw a type or attribute error by itself if there is a wrong type